client/client.go

```diff
@@ -402,7 +406,49 @@ func (client *Client) doSync(method, path string, query url.Values, headers map[
 	return client.doSyncWithOpts(method, path, query, headers, body, v, nil)
 }
 
+func (client *Client) checkMaintenanceJSON() error {
+	f, err := os.Open(dirs.SnapdMaintenanceFile)
+	// just continue if we can't read the maintenance file and
+	// it does exist (i.e. perm error)
+	if err == nil {
+		// we have a maintenance file, try to read it
+		maintenance := &Error{}
+
+		if err := json.NewDecoder(f).Decode(&maintenance); err != nil {
+			// TODO: should we just let the request go through even if the
```

Comment:
This is not a problem if we check only on actual error from connecting to the socket.

Reply:
I think  it's probably easier to just not return errors from this function and indeed ignore malformed maintance.json. Or at most log them.